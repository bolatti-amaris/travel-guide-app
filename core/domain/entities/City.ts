export interface City {
    id: number;
    key: string;
    name: string;
    nativeName: string;
    currency: string;
    language: string;
  }