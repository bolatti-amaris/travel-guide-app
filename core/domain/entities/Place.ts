export interface Place {
  type: 'monument' | 'restaurant';
  name: string;
  coordinates: [number, number];
}
