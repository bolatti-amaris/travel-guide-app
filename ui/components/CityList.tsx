import React from 'react';
import { FlatList } from 'react-native';
import CityCard from './CityCard';
import { City } from '@/core/domain/entities/city';

interface CityListProps {
  cities: City[];
}

const CityList = ({ cities }: CityListProps) => {
  return (
    <FlatList
      data={cities}
      keyExtractor={(item) => item.key}
      renderItem={({ item }) => (
        <CityCard
          {...item}
        />
      )}
    />
  );
};

export default CityList;
