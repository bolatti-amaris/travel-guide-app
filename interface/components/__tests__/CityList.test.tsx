import React from 'react';
import { render } from '@testing-library/react-native';
import CityList from '../CityList';
import { City } from '@/core/domain/entities/City';

jest.mock('../CityCard', () => {
  const { View } = require('react-native');
  return ({ city }: { city: City }) => (
    <View testID={`city-card-${city.key}`} />
  );
});

const mockCities: City[] = [
  {
    id: 1,
    key: 'paris',
    name: 'Paris',
    nativeName: 'Paris',
    currency: 'EUR',
    language: 'fr',
  },
  {
    id: 2,
    key: 'barcelona',
    name: 'Barcelona',
    nativeName: 'Barcelona',
    currency: 'EUR',
    language: 'es',
  },
];

describe('CityList', () => {
  it('renders one CityCard per city in the list', () => {
    const { getByTestId } = render(<CityList cities={mockCities} />);
    expect(getByTestId('city-card-paris')).toBeTruthy();
    expect(getByTestId('city-card-barcelona')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(<CityList cities={mockCities} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
