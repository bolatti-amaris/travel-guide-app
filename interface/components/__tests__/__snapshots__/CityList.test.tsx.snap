// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList matches snapshot 1`] = `
<RCTScrollView
  data={
    [
      {
        "currency": "EUR",
        "id": 1,
        "key": "paris",
        "language": "fr",
        "name": "Paris",
        "nativeName": "Paris",
      },
      {
        "currency": "EUR",
        "id": 2,
        "key": "barcelona",
        "language": "es",
        "name": "Barcelona",
        "nativeName": "Barcelona",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        testID="city-card-paris"
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        testID="city-card-barcelona"
      />
    </View>
  </View>
</RCTScrollView>
`;
